x-pack/plugins/reporting/export_types/printable_pdf/server/lib/layouts/layout_factory.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { KbnServer, Size } from '../../../../../types';
+import { LayoutTypes } from '../../../common/constants';
+import { Layout } from './layout';
+import { PreserveLayout } from './preserve_layout';
+import { PrintLayout } from './print_layout';
+
+interface LayoutParams {
+  id: string;
+  dimensions: Size;
+}
+
+export function createLayout(server: KbnServer, layoutParams: LayoutParams): Layout {
```

Comment:
let's call this file `create_layout.ts` to name it after what its primary export is.